network/wsPeer.go

```diff
@@ -148,6 +150,17 @@ type wsPeer struct {
 
 	// createTime is the time at which the connection was established with the peer.
 	createTime time.Time
+	// connMonitor used to measure the relative performance of the connection
+	// compared to the other outgoing connections. Incoming connections would have this
+	// field set to nil.
+	connMonitor *connectionPerformanceMonitor
+
+	// peerMessageDelay is calculated by the connection monitor; it's the relative avarage per-message delay.
+	peerMessageDelay int64
```

Comment:
isn't the above `lastPingRoundTripTime` already what we want to measure?

Reply:
They measure different "creatures". One is the pure pinging time and one is the average relative message latency. Both are meaningful and important, and I think that we should add the ping time as well. (maybe in the next PR, just to reduce the number of changes here ? )